SeaPkg/Docs/PlatformIntegration/PlatformIntegrationSteps.md

```diff
@@ -0,0 +1,223 @@
+
+1. Validate section alignment.
+1. Read the PE/COFF header into memory and read the first section, which provides the number of sections.
+1. Loop through each section of the image, copying over the sections if they have a SizeOfRawData > 0.
+1. The behavior differs between the DEBUG and RELEASE builds, as highlighted below.
+1. Load the Codeview information if present.
+1. Ensure the size of the image is correctly aligned.
+
+All these steps are done with the guidance of platform validation rules. For more information on the validation rules, please
+see the [SEA Validation Rules](#sea-validation-rules) section.
+
+## Platform Data Requirements
+
+The platform needs to produce the data structures in this section. The structures are consumed by MM Supervisor code to allow
+the `SmmCpuFeaturesLib` acquire platform-specific details.
+
+### HOBs Required by `SmmCpuFeaturesLib`
+
+1. `gMsegSmramGuid` - MSEG memory region information.
+   - [UefiCpuPkg/Include/Guid/MsegSmram.h](https://github.com/tianocore/edk2/blob/master/UefiCpuPkg/Include/Guid/MsegSmram.h)
+   - Note that [`MsegSmramPei`](../../Drivers/MsegSmramPei/MsegSmramPei.inf)
+   can be used to help produce this HOB.
+
+### SEA Validation Rules
+
+The SEA Validation Rule is a data structure used to guide the SEA core to validate certain data regions of MM supervisor
+core. Platform authors need to provide the validation rules during build time, which will be compiled and baked into the
+SEA core binary. The rules will be used to validate the state of the MM supervisor and revert the content of the MM supervisor
+before SEA attempts to de-relocate the MM supervisor for signature validation.
+
+As defined in this [header file](../../Include/SeaAuxiliary.h), the SEA core will use the following rules to validate the
+MM memory regions:
+
+1. `IMAGE_VALIDATION_ENTRY_TYPE_NONE`: Ignore the region corresponding to this entry.
+1. `IMAGE_VALIDATION_ENTRY_TYPE_MSEG`: The region is MSEG, SEA core will validate the region to be MSEG.
+1. `IMAGE_VALIDATION_ENTRY_TYPE_CONTENT`: The content inside this region matches exactly the content inside the rule.
+1. `IMAGE_VALIDATION_ENTRY_TYPE_MEM_ATTR`: The memory attribute of the region matches the rule.
+1. `IMAGE_VALIDATION_ENTRY_TYPE_SELF_REF`: The region is self-referenced, SEA core will validate the region to be pointing
+to another point inside the MM supervisor.
+1. `IMAGE_VALIDATION_ENTRY_TYPE_POINTER`: The entry is a pointer to and the pointer needs to be not null, pointing to non
+MSEG, or pointing to outside of TSEG, depending on the rule.
+
+The syntax of the rules is as described in this [documentation](../../Tools/GenSeaArtifacts/gen_aux/readme.md).
+
+## SEA Code Integration
+
+A general description of SEA and MM supervisor code integration is depicted below:
+
+![SEA and MM Supervisor Assert Layout](Images/asset_layout.png)
+
+1. Ensure all submodules for the platform are based on the latest Project Mu version (e.g. "202102")

```

Comment:
Probably should use the most up-to-date mu verison.

Reply:
Especially because SEA wouldn't work with 202102.

Reply:
Updated in latest push